vng_api_common/audittrails/viewsets.py

```diff
@@ -0,0 +1,123 @@
+            bron=self.audit.component_name,
+            actie=action,
+            actieWeergave=AuditTrailAction.labels.get(action, ''),
+            resultaat=status_code,
+            hoofdObject=main_object,
+            resource=self.basename,
+            resourceUrl=data['url'],
+            oud=version_before_edit,
+            nieuw=version_after_edit,
+        )
+        trail.save()
+
+
+class AuditTrailCreateMixin(AuditTrailMixin):
+    def create(self, request, *args, **kwargs):
+        response = super().create(request, *args, **kwargs)
+        self.create_audittrail(
+            response.status_code,
+            AuditTrailAction.create,
+            version_before_edit=None,
+            version_after_edit=response.data,
+        )
+        return response
+
+
+class AuditTrailUpdateMixin(AuditTrailMixin):
+    def update(self, request, *args, **kwargs):
+        # Retrieve the data stored in the object before updating
+        instance = self.get_object()
+        serializer = self.get_serializer(instance)
+        version_before_edit = serializer.data
+
+        action = AuditTrailAction.partial_update if kwargs.get('partial', False) else AuditTrailAction.update
+
+        response = super().update(request, *args, **kwargs)
+        self.create_audittrail(
+            response.status_code,
+            action,
+            version_before_edit=version_before_edit,
+            version_after_edit=response.data,
+        )
+        return response
+
+
+class AuditTrailDestroyMixin(AuditTrailMixin):
+    def destroy(self, request, *args, **kwargs):
+        # Retrieve the data stored in the object before updating
+        instance = self.get_object()
+        serializer = self.get_serializer(instance)
+        version_before_edit = serializer.data
+
+        # If the resource being deleted is the main resource, delete all the
+        # audittrails associated with it
+        if self.basename == self.audit.main_resource:
+            with transaction.atomic():
+                response = super().destroy(request, *args, **kwargs)
+                self._destroy_related_audittrails(version_before_edit['url'])
+                return response
+        else:
+            response = super().destroy(request, *args, **kwargs)
+            self.create_audittrail(
+                response.status_code,
+                AuditTrailAction.delete,
+                version_before_edit=version_before_edit,
+                version_after_edit=None
+            )
+            return response
+
+    def _destroy_related_audittrails(self, main_object_url):
+        AuditTrail.objects.filter(hoofdObject=main_object_url).delete()
+
+
+class AuditTrailViewsetMixin(AuditTrailCreateMixin,
+                             AuditTrailUpdateMixin,
+                             AuditTrailDestroyMixin):
+    pass
+
+
+class AuditTrailViewset(viewsets.ReadOnlyModelViewSet, NestedViewSetMixin):
+    queryset = AuditTrail.objects.all()
```

Comment:
Can you put a docstring here that shows how to use this `AuditTrailViewSet` (capital S btw) in components?